### PR TITLE
Fix PID instrumentation for configuration caching

### PR DESCRIPTION
**Pull request by wolfs**
Use a build service for PID instrumentation on Gradle versions >= 6.1, so the PID collector is
compatible with configuration caching.

### CODE REVIEW

File: subprojects/build-operations/src/main/java/org/gradle/trace/pid/PidCollectorBuildService.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    public interface Parameters extends BuildServiceParameters {
+        RegularFileProperty getPidFile();
+    }
+
+    public PidCollectorBuildService() throws Exception {
+        GFileUtils.writeFile(getPid().toString(), getParameters().getPidFile().get().getAsFile());
+    }
+
+    private Integer getPid() throws NoSuchFieldException, IllegalAccessException, NoSuchMethodException, InvocationTargetException {
```
Comment (wolfs) [inline subprojects/build-operations/src/main/java/org/gradle/trace/pid/PidCollectorBuildService.java:35]: I am not very happy with how we obtain the PID of the current process here, though I don't have a better idea :(.

File: src/test/groovy/org/gradle/profiler/AbstractProfilerIntegrationTest.groovy [hunk 2]
```diff
@@ -16,7 +16,7 @@ abstract class AbstractProfilerIntegrationTest extends Specification {
         "3.3", "3.4.1", "3.5",
         "4.0", "4.1", "4.2.1", "4.7",
         "5.2.1", "5.5.1", "5.6.3",
-        "6.0.1", "6.1", "6.6-rc-2"
+        "6.0.1", "6.1", "6.6"
```
Comment (blindpirate) [inline src/test/groovy/org/gradle/profiler/AbstractProfilerIntegrationTest.groovy:19]: `6.6.1`?

Comment (wolfs) [reply src/test/groovy/org/gradle/profiler/AbstractProfilerIntegrationTest.groovy]: Done. I also upgradled the build to 6.6.1.

File: subprojects/build-operations/src/main/java/org/gradle/trace/pid/PidCollectorBuildService.java (hunk 1 above)
Comment (blindpirate) [reply subprojects/build-operations/src/main/java/org/gradle/trace/pid/PidCollectorBuildService.java:35]: I guess no better solution on Java 8.

**Review by blindpirate — APPROVED** [review]
LGTM, thanks for fixing!

**Pull request merged**